### ISSUE TITLE
[TASK] agent: refactor the finalize_slaves_connect_state

### ISSUE THREAD

**Issue by rmelotte** [prplfoundation/prplMesh#911]
## Description

Follow-up from PR [#891](https://github.com/prplfoundation/prplMesh/pull/891), specifically https://github.com/prplfoundation/prplMesh/pull/891#issuecomment-593910387

 - Further refactor the finalize_slaves_connect_state, split it into two functions for the connect and disconnect branches. They're completely disjoint. Actually, since each function is called only once, they could be inlined, but they're rather big so let's keep them as functions.
  - After the last commit, pSocket will always be non-nullptr, so the looping over the different sockets becomes redundant. I would then move the loop over socs that is currently in handle_1905_autoconfiguration_response down into finalize_slaves_connect_state, and just pass the freq_type as argument instead of pSocket.
